src/interceptor.ts

```diff
@@ -44,6 +45,11 @@ export const aws4Interceptor = (options?: InterceptorOptions, credentials?: Cred
     throw new Error("No URL present in request config, unable to sign request");
   }
 
+  if(config.params) {
+    config.url = buildUrl(config.url, config.params, config.paramsSerializer);
```

Comment:
I think mutating an incoming `config` is an action at a distance anti-pattern. Better to make a copy (deep clone).

Reply:
Not sure if I can follow regarding the anti-pattern. The params object would need to be deleted. I can create a clone at the beginning and refactor all following operations using the cloned config. But it adds overhead, if people use more complex custom serializer functions and stuff.

Reply:
As a package consumer, I do not expect the package to make any modifications to my input object.

Imagine I'd want to make two queries, with small modifications:

```ts
const req = {
  method: "GET",
  params: {
    foo: 1
  }
}

request(req)

// this request would fail, because `req.params` was clobbered
req.params.foo = 2
request(req)
```

Reply:
You are right! Indeed, axios internals make a deep clone already.